packages/react-workers/__tests__/sorting.test.js

```diff
@@ -0,0 +1,79 @@
+import { applySorting } from '../src/utils/sorting';
+
+const features = [
+  { column1: 'C', column2: 3 },
+  { column1: 'A', column2: 1 },
+  { column1: 'B', column2: 2 },
+  { column1: 'D', column2: 4 },
+  { column1: 'D', column2: 5 }
+];
+
+const commonSortedFeatures = [
+  { column1: 'A', column2: 1 },
+  { column1: 'B', column2: 2 },
+  { column1: 'C', column2: 3 },
+  { column1: 'D', column2: 4 },
+  { column1: 'D', column2: 5 }
+];
+
+const commonSortedFeatures2 = [
+  { column1: 'D', column2: 4 },
+  { column1: 'D', column2: 5 },
+  { column1: 'C', column2: 3 },
+  { column1: 'B', column2: 2 },
+  { column1: 'A', column2: 1 }
+];
+
+describe('Sorting', () => {
+  describe('should correctly understand sortOptions', () => {
+    test('if undefined', async () => {
+      expect(applySorting(features)).toEqual(features);
+    });
+
+    test('if sortBy is string', async () => {
+      expect(applySorting(features, { sortBy: 'column1' })).toEqual(commonSortedFeatures);
+    });
+
+    test('if sortBy is array of string', async () => {
+      expect(applySorting(features, { sortBy: ['column1', 'column2'] })).toEqual(
+        commonSortedFeatures
+      );
+    });
+
+    test('if sortBy is array of arrays', async () => {
+      expect(applySorting(features, { sortBy: [['column1'], ['column2']] })).toEqual(
+        commonSortedFeatures
+      );
+    });
+  });
+  describe('should correctly sort', () => {
+    test('if sortByDirection is used', async () => {
+      expect(
+        applySorting(features, { sortBy: 'column1', sortByDirection: 'desc' })
+      ).toEqual(commonSortedFeatures2);
+    });
+
+    test('if sort direction is applied inside sortBy', async () => {
+      expect(applySorting(features, { sortBy: [['column1', 'desc']] })).toEqual(
+        commonSortedFeatures2
+      );
+
+      expect(
+        applySorting(features, { sortBy: [['column1']], sortByDirection: 'desc' })
+      ).toEqual(commonSortedFeatures2);
+    });
+
+    test('if sort direction is applied inside sortBy and sortByDirection is also used, sortBy has priority', async () => {
+      expect(
+        applySorting(features, { sortBy: [['column1', 'desc']], sortByDirection: 'asc' })
```

Comment:
I see this is really flexible, with quite a lot of overloads / combinations to pass options... I'm just wondering if that could cause issues and should be just simplified, what do you think?

Reply:
I think we shouldn't be afraid of that because there isn't many tricks that can be done, and all the possible tricks are resolved in one function (normaliseOptions if my memory doesn't lie), then if there's any bug, we know where we can find them, then I wouldn't be worried about that, I think it's pretty controlled.